### PR TITLE
Add Feature Dependencies changes to main spec

### PR DESCRIPTION
**Pull request by joshspicer**
ref https://github.com/devcontainers/spec/issues/109

Synthesizes the Feature Dependencies proposal (https://github.com/devcontainers/spec/pull/234) into the specification

### CODE REVIEW

File: docs/specs/devcontainer-features.md [hunk 1]
```diff
@@ -260,45 +263,162 @@ To ensure that the appropriate shell is used, the execute bit should be set on `
...
+
+In the snippet above, `myfeature` MUST be installed after `foo`, `bar`, and `baz`.  If the Features provided via the `dependsOn` property declare their own dependencies, those must also be satisfied before the Feature is installed.
+
+#### The `installsAfter` Property
+
+The `installsAfter` property indicates a "soft dependency" that influences the installation order of Features that are already queued to be installed.  The effective behavior of this property is the same as `dependsOn`, with the following differences:
+
+- `installsAfter` is **not** evaluated recursively.
+- `installsAfter` only influences the installation order of Features that are **already set to be installed**.  Any Feature not set to be installed after (1) resolving the `dependsOn` dependency tree or (2) indicated by the user's `devcontainer.json` should not be added to the installation list.
+- The Feature indicated by `installsAfter` can **not** provide options, nor are they able to be pinned to a specific version tag or digest.  Resolution to the canonical name should still be performed (eg: If the Feature has been [renamed](#steps-to-rename-a-feature)).
```
Comment (samruddhikhandale) [inline docs/specs/devcontainer-features.md:303]: > The Feature indicated by `installsAfter` can **not** provide options, nor are they able to be pinned to a specific version tag or digest

Should we respecify and highlight this point under `dependsOn` property section? It's amazing to be able to specify options ❇️

File: docs/specs/devcontainer-features.md [hunk 2]
```diff
@@ -260,45 +263,162 @@ To ensure that the appropriate shell is used, the execute bit should be set on `
...
+> 
+> Repeat for as many rounds as necessary until `worklist` is empty.  If there is ever a round where no elements are added to `installationOrder`, the algorithm should terminate and return an error.  This indicates a circular dependency or other fatal error in the dependency graph.  Implementations should attempt to  provide the user with information about the error and possible mitigation strategies.
+>
+> ### Notes
+>
+> From an implementation point of view, `installsAfter` nodes may be added as a seperate set of directed edges, just as `dependsOn` nodes are added as directed edges (see **(1)**).  Before round-based installation and sorting **(3)**, an orchestrating tool should remove all `installsAfter` directed edges that do not correspond with a Feature in the `worklist` that is set to be installed.  In each round, a Feature can then be installed if all its requirements (both `dependsOn` and `installsAfter` dependencies) have been fulfilled in previous rounds.
```
Comment (samruddhikhandale) [inline docs/specs/devcontainer-features.md:417]:
```suggestion
> From an implementation point of view, `installsAfter` nodes may be added as a separate set of directed edges, just as `dependsOn` nodes are added as directed edges (see **(1)**).  Before round-based installation and sorting **(3)**, an orchestrating tool should remove all `installsAfter` directed edges that do not correspond with a Feature in the `worklist` that is set to be installed.  In each round, a Feature can then be installed if all its requirements (both `dependsOn` and `installsAfter` dependencies) have been fulfilled in previous rounds.
```

**Review by samruddhikhandale — APPROVED** [review]
Exciting!!!! 🚀 🚀 
Thanks for writing this up! ❇️

File: docs/specs/devcontainer-features.md (hunk 1 above)
Comment (joshspicer) [reply docs/specs/devcontainer-features.md:305]: I've more explicitly pointed out the ability to provide options in this section https://github.com/devcontainers/spec/pull/292/commits/acdfd76a0fd3813b38131277614ed40609c01878

**Pull request merged**